rbtcwrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -0,0 +1,233 @@
+     * @param  _poolTokenAmount    minted pool token amount
+     */
+    event LiquidityAdded(
+        address indexed _provider,
+        uint256 indexed _reserveAmount,
+        uint256 indexed _poolTokenAmount
+    );
+
+    /**
+     * @dev triggered after liquidity is removed
+     *
+     * @param  _provider           liquidity provider
+     * @param  _reserveAmount      added reserve token amount
+     * @param  _poolTokenAmount    burned pool token amount
+     */
+    event LiquidityRemoved(
+        address indexed _provider,
+        uint256 indexed _reserveAmount,
+        uint256 indexed _poolTokenAmount
+    );
+
+    /**
+     * @dev triggered after liquidity is removed
+     *
+     * @param _beneficiary                 account that will receive the conversion result or 0x0 to send the result to the sender account
+     * @param _sourceTokenAmount           amount to convert from, in the source token
+     * @param _targetTokenAmount           amount of tokens received from the conversion, in the target token
+     * @param _sovrynSwapNetworkAddress    address of SovrynSwapNetwork contract
+     * @param _path                        conversion path between two tokens in the network
+     */
+    event TokenConverted(
+        address indexed _beneficiary,
+        uint256 indexed _sourceTokenAmount,
+        uint256 indexed _targetTokenAmount,
+        address _sovrynSwapNetworkAddress,
+        IERC20Token[] _path
+    );
+
+    /**
+     * @dev To check if ddress is contract address 
+     */
+    modifier checkAddress(address address_) {
+        require(address_.isContract(), "The address is not a contract");
+        _;
+    }
+    
+    constructor(address _wrbtcTokenAddress) public checkAddress(_wrbtcTokenAddress) {
+        wrbtcTokenAddress = _wrbtcTokenAddress;
+        wrbtc = IWrbtcERC20(wrbtcTokenAddress);        
+    }
+
+    function() external payable { }
+
+    /**
+     * @dev  
+     * The process:
+     * 1.Accepts RBTC
+     * 2.Sends RBTC to WRBTC contract in order to wrap RBTC to WRBTC
+     * 3.Calls 'addLiquidity' on LiquidityPoolConverter contract
+     * 4.Transfers pool tokens to user
+     * 
+     * @param _liquidityPoolConverterAddress    address of LiquidityPoolConverter contract
+     * @param _amount                           amount of liquidity to add
+     * @param _minReturn                        minimum return-amount of reserve tokens
+     *
+     * @return amount of pool tokens minted
+     */
+    function addLiquidity(
+        address _liquidityPoolConverterAddress, 
+        uint256 _amount, 
+        uint256 _minReturn
+    )  
+        public
+        payable
+        checkAddress(_liquidityPoolConverterAddress)
+        returns(uint256) 
+    {
+        require(_amount == msg.value, "The provided amount should be identical to msg.value");
+
+        liquidityPoolConverter = ILiquidityPoolV2Converter(_liquidityPoolConverterAddress);
```

Comment:
now, you're still writing to storage, Wukong.
There's no need to write the address to storage. this only consumes a lot of gas. 
Please use a local variable instead.